app/src/main/java/com/github/se/eventradar/viewmodel/EventsOverviewViewModel.kt

```diff
@@ -30,6 +36,130 @@ constructor(
     checkUserLoginStatus()
   }
 
+  fun onSearchQueryChanged(
```

Comment:
**Question:** Why is _uiState passed as a default parameter? It doesn't seem necessary. 

Note: This question applies to all functions in the file that use _uiState as a default parameter.

Reply:
It means that when the function is called in the UI (view), it doesn't need to pass in any parameters. We know that the UI will always act on the same viewmodel, explicitly the default UiState specified in the EventsOverviewViewModel. So it is cleaner in the UI to pass _uiState as a default parameter.

Reply:
I meant: Why not directly use the _uiSate without passing it as a parameter, since we are already inside EventsOverviewVM?

Reply:
It means you don't have to pass it as a parameter in the UI